gguf-py/tests/test_gguf.py

```diff
@@ -1,7 +1,48 @@
-import gguf
+import os
+import tempfile
+import pytest
+import numpy as np
+from gguf import GGUFWriter, GGMLQuantizationType, GGUFValueType, MODEL_ARCH
 
-# TODO: add tests
+# Fixture to create a temporary directory for testing
+@pytest.fixture
+def temp_dir():
+    with tempfile.TemporaryDirectory() as tempdir:
+        yield tempdir
 
+# Test GGUFWriter class
+def test_gguf_writer(temp_dir):
+    # Create a GGUFWriter instance
+    writer = GGUFWriter(os.path.join(temp_dir, "test.gguf"), "gpt2")
 
-def test_write_gguf():
-    pass
+    # Add metadata
+    writer.add_name("Test Model")
+    writer.add_author("John Doe")
+    writer.add_description("A test model")
+    writer.add_architecture()
+    writer.add_tensor_data_layout("NHWC")
+    writer.add_context_length(512)
+
+    # Add a tensor
+    tensor_data = np.random.rand(10, 10).astype(np.float32)
+    writer.add_tensor("test_tensor", tensor_data)
+
+    # Write to file and close
+    writer.write_header_to_file()
+    writer.write_tensors_to_file()
+    writer.close()
+
+    # Verify the generated file exists
+    assert os.path.exists(os.path.join(temp_dir, "test.gguf"))
```

Comment:
Should we also verify the contents of the file? You could either create this actual file as a  gguf with the expected contents and check it in then compare whatever file we write with the snapshot, or you could use a gguf reader and pull out all the keys and assert they have the right values. That will guard against inadvertent changes to the format etc